### PR TITLE
fix: add headers for file download

### PR DESCRIPTION
**Pull request by vkrashenko**
## Description
The token is not passed when user downloading the file from quote
### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-931
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.56.0-pr-1042-751e-751e3d06.zip

### CODE REVIEW

File: client-app/shared/files/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useAuth } from "@/core/composables";
+
+const { headers } = useAuth();
+export async function generateLinkAndDownloadFile(fileUrl: string, fileName: string) {
+  try {
+    const response = await fetch(fileUrl, {
```
Comment (NaMax66) [inline client-app/shared/files/utils/index.ts:6]: use client-app/core/api/common/composables/useFetch.ts

there is interceptor inside which add the header

File: client-app/shared/files/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useAuth } from "@/core/composables";
+
+const { headers } = useAuth();
+export async function generateLinkAndDownloadFile(fileUrl: string, fileName: string) {
```
Comment (NaMax66) [inline client-app/shared/files/utils/index.ts:4]: use function name downloadFile

File: client-app/ui-kit/components/molecules/file/vc-file.vue [hunk 3]
```diff
@@ -12,9 +12,9 @@
 
     <span class="vc-file__details">
       <span class="vc-file__row">
-        <a v-if="isAttachedFile(file) || isUploadedFile(file)" class="vc-file__link" :href="file.url" target="_blank">
+        <span v-if="isAttachedFile(file) || isUploadedFile(file)" class="vc-file__link" @click="downloadFile(file)">
```
Comment (NaMax66) [inline client-app/ui-kit/components/molecules/file/vc-file.vue:15]: try to keep <a> for accessibility. Just @click.prevent should help

**Review by NaMax66 — CHANGES_REQUESTED** [review]

File: client-app/shared/files/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    const { data } = await useFetch(fileUrl).blob();
+    const url = window.URL.createObjectURL(toValue(data) as Blob);
+    const link = document.createElement("a");
+    link.href = url;
+    link.setAttribute("download", fileName);
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+  } catch (error) {
+    console.error("Request error:", error);
```
Comment (Andrew-Orlov) [inline client-app/shared/files/utils/index.ts]: We have Logger

**Review by NaMax66 — APPROVED** [review]

**Pull request merged**